docker/puppetdb/Dockerfile

```diff
@@ -56,4 +56,15 @@ EXPOSE 8080 8081
 ENTRYPOINT ["dumb-init", "/docker-entrypoint.sh"]
 CMD ["foreground"]
 
+# The start-period is just a wild guess how long it takes PuppetDB to come
+# up in the worst case. The other timing parameters are set so that it
+# takes at most a minute to realize that PuppetDB has failed.
+HEALTHCHECK --interval=10s --timeout=10s --retries=6 \
+  --start-period=5m CMD \
+  curl --fail --silent \
+  --resolve 'puppetdb:8080:127.0.0.1' \
+  http://puppetdb:8080/status/v1/services/puppetdb-status \
+  | grep -q '"state":"running"' \
+  || exit 1
+
```

Comment:
I can't assume jq (or don't want to require having that installed in the container), but I like your Python idea. `grep -v` won't work since that would just list the lines that do not have the state line in them.

Reply:
I believe `grep -v` also inverts the exit status.

Reply:
Wait, ignore the `-v` comment, I'd confused myself -- if the goal is for the pipeline to only succeed if we find "running", then actually I don't think we need the `|| exit 1` since `grep -q` will already exit with status 0 or 1 (not 2, because of the `-q`).  (I now see that the `exit 1` is probably (normally) there because docker requires only 0 or 1).  But in any case, not a big deal either way.

Reply:
What does this `puppetdb-status` call do? I'm new to PuppetDB, so I want to make sure it's not an expensive operation since it's going to happen every 10 seconds.

Reply:
@robdaemon it's just hitting this: https://puppet.com/docs/puppetdb/6.0/api/status/v1/status.html